### PR TITLE
Concrete Wet/Wall Code Cleanup

### PR DESCRIPTION
**Pull request by MarkNais**
## About the PR 
Refactors repeated code into a proc call, and makes the conrete wall a little more robust incase it's spawned **NOT** form wet concrete.
Also fixed `CanPass()` override. It wasn't properly calling the parent / allowing users to enter the turf.

Was reading up on Concrete since it wasn't in the wiki and isn't meant to be secret.

## Why's this needed? 
Cleaner code better code.
Fixes bug.

[CHORE][BUG]

### CODE REVIEW

File: code/obj/cement.dm [hunk 1]
```diff
@@ -68,7 +61,7 @@
 	flags = FPRINT | CONDUCT | USEDELAY
 	var/strength = 0 // 1=poor, 2=ok, 3=good, 4=perfect
 	var/health = 30 //health num modified in New, 30 for poor, 60 for ok, 90 for good, 120 for perfect
```
Comment (ZeWaka) [inline code/obj/cement.dm:63]: doesn't this var already exist on all objects? _health or something?

File: code/obj/cement.dm [hunk 2]
```diff
@@ -79,8 +72,8 @@
 			loc:ReplaceWithConcreteFloor()
 
 		update_nearby_tiles(1)
-		SPAWN_DBG(1)
-			RL_SetOpacity(1)
+		sleep(1)
```
Comment (ZeWaka) [inline code/obj/cement.dm:75]:
```suggestion
		sleep(0.1 SECONDS)
```

**Review by ZeWaka — COMMENTED** [review]
also i'm pretty sure it's meant to be secret

**Comment by Adharainspace** [comment]
it isnt. i thought itd be neat but then i realised that its kinda underwhelming rn. no worries abt it being secret.

File: code/obj/cement.dm (hunk 1 above)
Comment (MarkNais) [reply code/obj/cement.dm]: Yep, refactored to use it now.

**Comment by Tiggersaurus** [comment]
> also i'm pretty sure it's meant to be secret

I was under the impression it was secret too ...

**Comment by MarkNais** [comment]
Adhara spoke on the discord: it was initially meant to be secret, but she's since changed her mind.
(Not that this PR is changing that from a code perspective)

File: code/obj/cement.dm [hunk 3]
```diff
@@ -16,44 +18,32 @@
...
-		if (health <= 0)
-			user.visible_message("<span class='alert'>[user] breaks apart the lump of wet concrete with their bare hands!</span>")
-			qdel(src)
-			return
-		health--
-		if (health <= 0)
-			user.visible_message("<span class='alert'>[user] breaks apart the lump of wet concrete with their bare hands!</span>")
-			qdel(src)
-			return
+		//changeHealth(-1)
```
Comment (ZeWaka) [inline code/obj/cement.dm:38]: useless proc now

File: code/obj/cement.dm [hunk 4]
```diff
@@ -6,7 +6,9 @@
 	density = 0
 	layer = OBJ_LAYER + 0.9
 	event_handler_flags = USE_CANPASS
-	var/health = 4 //how many hits to destroy + 1
+	var/const/initial_health = 4
```
Comment (ZeWaka) [inline code/obj/cement.dm:9]: why is one baseHealth and the other one initial_health

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/cement.dm (hunk 4 above)
Comment (ZeWaka) [inline code/obj/cement.dm:9]: i don't think this is a good value, as this is extremely weak compared to everything else on station.... see ex_act(3) throwing it to -40 or whatever

**Comment by ZeWaka** [comment]
@MarkNais marking this non-organic

if you want to finish this, that'd be great

**Comment by MarkNais** [comment]
Changes made. Increased initial health of the wet concrete to 10, results to 5 hits from a tool to break. Original intention was for wet to be broken easily.

**Comment by ZeWaka** [comment]
> Changes made.

What about the other two comments above?

**Comment by MarkNais** [comment]
I mis-read the first request, proc removed now.
Not sure what "comment #2" is.

**Comment by MarkNais** [comment]
I'm dumb, don't code while tired. Replying

File: code/obj/cement.dm (hunk 4 above)
Comment (MarkNais) [reply code/obj/cement.dm]: Just wanted to reuse the number instead of it being changed in one place and not being updated in another (code wise)

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**